### PR TITLE
feat(glcdc): screen rotation support

### PR DESCRIPTION
**Pull request by becseya**
This PR introduces screen rotation for GLCDC driver in partial mode.

### CODE REVIEW

File: src/drivers/display/renesas_glcdc/lv_renesas_glcdc.c [hunk 1]
```diff
@@ -277,13 +282,35 @@ static void flush_wait_direct(lv_display_t * display)
...
+    if(rotation != LV_DISPLAY_ROTATION_0) {
+        int32_t w = lv_area_get_width(area);
+        int32_t h = lv_area_get_height(area);
+        uint32_t w_stride = lv_draw_buf_width_to_stride(w, cf);
+        uint32_t h_stride = lv_draw_buf_width_to_stride(h, cf);
+
+        // only allocate if rotation is actually being used
+        if(!rotation_buffer)
+            rotation_buffer = lv_malloc(partial_buffer_size);
+        LV_ASSERT(rotation_buffer);
```
Comment (FASTSHIFT) [inline src/drivers/display/renesas_glcdc/lv_renesas_glcdc.c]: LV_ASSERT_MALLOC

**Comment by kisvegabor** [comment]
cc @Forman1998

**Review by kisvegabor — DISMISSED** [review]

File: src/drivers/display/renesas_glcdc/lv_renesas_glcdc.c [hunk 2]
```diff
@@ -76,6 +77,9 @@ static glcdc_runtime_cfg_t  g_layer_change;
 display_t g_display0_cfg;
 #endif /*_RENESAS_RX_*/
 
+static void * rotation_buffer = nullptr;
```
Comment (FASTSHIFT) [inline src/drivers/display/renesas_glcdc/lv_renesas_glcdc.c]: Will this compile with an error? `nullptr` is used in C++11, while C uses `NULL`.

**Review by kisvegabor — DISMISSED** [review]

File: src/drivers/display/renesas_glcdc/lv_renesas_glcdc.c [hunk 3]
```diff
@@ -277,13 +282,35 @@ static void flush_wait_direct(lv_display_t * display)
...
+    if(rotation != LV_DISPLAY_ROTATION_0) {
+        int32_t w = lv_area_get_width(area);
+        int32_t h = lv_area_get_height(area);
+        uint32_t w_stride = lv_draw_buf_width_to_stride(w, cf);
+        uint32_t h_stride = lv_draw_buf_width_to_stride(h, cf);
+
+        // only allocate if rotation is actually being used
+        if(!rotation_buffer)
+            rotation_buffer = lv_malloc(partial_buffer_size);
+        LV_ASSERT_MALLOC(rotation_buffer);
```
Comment (liamHowatt) [inline src/drivers/display/renesas_glcdc/lv_renesas_glcdc.c]: Maybe move this into the `if` branch.

File: src/drivers/display/renesas_glcdc/lv_renesas_glcdc.c [hunk 4]
```diff
@@ -277,13 +282,35 @@ static void flush_wait_direct(lv_display_t * display)
...
+
+        if(rotation == LV_DISPLAY_ROTATION_180)
+            lv_draw_sw_rotate(img, rotation_buffer, w, h, w_stride, w_stride, rotation, cf);
+        else /* 90 or 270 */
+            lv_draw_sw_rotate(img, rotation_buffer, w, h, w_stride, h_stride, rotation, cf);
+
+        img = rotation_buffer;
+        lv_display_rotate_area(display, (lv_area_t *)area);
```
Comment (liamHowatt) [inline src/drivers/display/renesas_glcdc/lv_renesas_glcdc.c]: Since it can be avoided, maybe copy `area` and modify that instead.

**Review by liamHowatt — COMMENTED** [review]
If my own logging verbosity were turned down one level, I wouldn't have any comments :wink:

**Review by kisvegabor — APPROVED** [review]

**Comment by becseya** [comment]
Please do not squash when merging

**Review by liamHowatt — APPROVED** [review]

**Pull request merged**